lib/src/metta/text.rs

```diff
@@ -101,13 +101,13 @@ impl<'a> SExprParser<'a> {
         }
     }
 
-    fn parse_atom(&mut self, tokenizer: &Tokenizer) -> Option<Atom> {
-        let token = next_token(&mut self.it);
+    fn parse_atom(&mut self, tokenizer: &Tokenizer) -> Result<Option<Atom>, String> {
```

Comment:
I know you just kept the previous API, but it looks like `Option` is not needed here, so the return result could be just `Result<Atom, String>`.